src/core/qgscoordinatetransform.cpp

```diff
@@ -821,7 +821,10 @@ bool QgsCoordinateTransform::setFromCache( const QgsCoordinateReferenceSystem &s
   if ( sourceKey.isEmpty() || destKey.isEmpty() )
     return false;
 
-  sCacheLock.lockForRead();
+  QgsReadWriteLocker locker( sCacheLock, QgsReadWriteLocker::Read );
```

Comment:
Could this be moved to after the condition to avoid the locking/unlocking in case caching is disabled?

Reply:
I'm trying to be super-safe here and also protect the disable cache static boolean value.

Reply:
Hmm... ok.
I wouldn't expect issues (if there's a concurrent write at the same time changing the value the result will be a matter of timing with or without locking). But yes it still feels slightly cleaner.

Reply:
I'm still inclined to be overly safe here, especially given that the cache is only ever disabled on shutting down qgis, and we would ideally expect this situation to never arise (accessing cache during the shutdown process).

Reply:
All good :)

Reply:
Let's just hope this finally puts this issue to bed...

Reply:
fingers crossed!

Reply:
Going to a nearby pagoda to light up so incense! ;)